llvm/lib/Target/NVPTX/NVPTXInstrInfo.td

```diff
@@ -2637,25 +2637,49 @@ class LoadParamRegInst<NVPTXRegClass regclass, string opstr> :
                 [(set regclass:$dst, (LoadParam (i32 0), (i32 imm:$b)))]>;
 
 let mayStore = true in {
-  class StoreParamInst<NVPTXRegClass regclass, string opstr> :
+  class StoreParamInstReg<NVPTXRegClass regclass, string opstr> :
```

Comment:
I'd move the implementation into `def _r:` below, and used `...._r` instead for `StoreParamI8TruncI32`.

Reply:
Sorry, I'm not sure I follow this one, can you clarify what you mean?

Reply:
Down below you have `StoreParamInst` which defines register and immediate store ops. 
`StoreParamInstReg` provides only a register-based store.
I propose to keep instruction definitions within one class only, in `StoreParamInst`, and eliminate `StoreParamInstReg` as redundant.

Reply:
If I were to do this how would I define the `StoreParamI8TruncI32` and `StoreParamI8TruncI64` instructions for which I do not want a `_i` variant?

Reply:
Considering that those `_i` and `_r` records carry no pattern, having an unused record will not hurt, it will remain unused, unless someone explicitly refers to those records. If you want to make 100% sure it's never used, you can add an argument, and add an always-false predicate to the `_i` variant if it's set.

If it's too much of a hassle, the code is OK as is, functionally, just looks a bit odd to me.  I would expect _i and _r to be defined very similarly, yet one is defined directly, and another via a separate class. It kind of begs to be straightened out -- define _i and _r the same way, either directly, or via the same external class, parametrized accordingly. Direct use of NVPTXInst looks the simplest choice here. Defining a few unused instruction variants is OK -- we have tons of them, already, due to the quirkiness of tensorcore instructions.

I'll leave it up to you.

Reply:
Thanks for explaining, I've added an argument as you describe but with an `if` to prevent the `_i` variants from being defined. I agree it is cleaner to unify the definitions as they are identical.